src/Dapr.Jobs/Extensions/EndpointRouteBuilderExtensions.cs

```diff
@@ -0,0 +1,75 @@
+// limitations under the License.
+// ------------------------------------------------------------------------
+
+using System.Text.Json;
+using Dapr.Jobs.Models.Responses;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Routing;
+
+namespace Dapr.Jobs.Extensions;
+
+/// <summary>
+/// Provides extension methods to register endpoints for Dapr Job Scheduler invocations.
+/// </summary>
+public static class EndpointRouteBuilderExtensions
+{
+    /// <summary>
+    /// Provides for a handler to be provided that allows the user to dictate how various jobs should be handled without
+    /// necessarily knowing the name of the job at build time.
+    /// </summary>
+    /// <param name="endpoints">The <see cref="IEndpointRouteBuilder"/> to add the route to.</param>
+    /// <param name="action">The asynchronous action provided by the developer that handles any inbound requests. The first two
+    /// parameters must be a nullable <see cref="string"/> for the jobName and a nullable <see cref="DaprJobDetails"/> with the
+    /// payload details, but otherwise can be populated with additional services to be injected into the delegate.</param>
+    /// <param name="cancellationToken">Cancellation token that will be passed in as the last parameter to the delegate action.</param>
+    public static IEndpointRouteBuilder MapDaprScheduledJobHandler(this IEndpointRouteBuilder endpoints,
+        Delegate action, CancellationToken cancellationToken = default)
+    {
+        ArgumentNullException.ThrowIfNull(endpoints, nameof(endpoints));
+        ArgumentNullException.ThrowIfNull(action, nameof(action));
+
+        endpoints.MapPost("/job/{jobName}", async context =>
+        {
+            var jobName = (string?)context.Request.RouteValues["jobName"];
+            DaprJobDetails? jobPayload = null;
+
+            if (context.Request.ContentLength is > 0)
+            {
+                using var reader = new StreamReader(context.Request.Body);
+                var body = await reader.ReadToEndAsync();
+
+                try
+                {
+                    var deserializedJobPayload = JsonSerializer.Deserialize<DeserializableDaprJobDetails>(body);
+                    jobPayload = deserializedJobPayload?.ToType() ?? null;
+                }
+                catch (JsonException)
+                {
+                    jobPayload = null;
+                }
+            }
+
+            var parameters = new List<object?> { jobName, jobPayload };
+            var actionParameters = action.Method.GetParameters().Skip(2).ToArray();
+            parameters.AddRange(actionParameters
+                .Where(parameter => parameter.ParameterType != typeof(CancellationToken))
+                .Select(parameter => context.RequestServices.GetService(parameter.ParameterType)));
+            parameters.Add(cancellationToken);
+            
+            var result = action.DynamicInvoke(parameters.ToArray());
```

Comment:
Snazzy, but could result in strange runtime errors if/when the user does not abide by the assumptions of the parameter order.  Would it be not much more difficult to allow *any* order (as is generally allowed by ASP.NET)?

E.g.:

```csharp
var parameters =
    action
        .Method
        .GetParameters()
        .Select(
            parameter => parameter.ParameterType switch
            {
                System.String => jobName,
                DaprJobDetails => daprJobDetails,
                CancellationToken => cancellationToken,
                _ => context.RequestServices.GetService(parameter.ParameterType)
            }
        .ToArray();
```

Reply:
Good point - updated to accommodate any order, including if the cancellation token is not provided by the user in their delegate (added as CancellationToken.None)